collector/src/compare.rs

```diff
@@ -10,11 +10,29 @@ use tabled::{Table, Tabled};
 /// Compare 2 artifacts and print the result.
 pub async fn compare_artifacts(
     mut conn: Box<dyn Connection>,
-    base: String,
-    modified: String,
+    base: Option<String>,
+    modified: Option<String>,
 ) -> anyhow::Result<()> {
     let index = database::Index::load(&mut *conn).await;
 
+    let aids = index.artifacts().collect::<Vec<_>>();
+    if aids.len() < 2 {
+        return Err(anyhow::anyhow!("There's no artifacts to compare."));
+    }
+
+    let select_artifact_id = |name: &str| {
+        anyhow::Ok(
+            inquire::Select::new(
+                &format!("Choose {} artifact to compare:", name),
+                aids.clone(),
+            )
+            .prompt()?,
+        )
+    };
+
+    let base = base.unwrap_or(select_artifact_id("base")?.to_string());
+    let modified = modified.unwrap_or(select_artifact_id("modified")?.to_string());
```

Comment:
It would be nice to remove the selected `base` artifact from the list offered in the select, so that you don't accidentally select the same artifact twice.